### PR TITLE
Add pydantic evented model

### PR DESCRIPTION
**Pull request by sofroniewn**
# Description
Following the suggestion of @tlambert03 in https://github.com/napari/napari/pull/2042#issuecomment-762498792 this would add the pydantic evented model unhooked up for testing before starting to use in more PRs like #2042, #2066 and others.

We should add some comprehensive tests before merge too.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

**Comment by sofroniewn** [comment]
@tlambert03 feel free to add some tests/ make edits to this as you see fit. I should have some more time tomorrow to work on it too. Would be nice to get it ready for review soon!!

**Comment by tlambert03** [comment]
I probably won't have time tomorrow, but what I would do (in addition to just the tests that seem right) is install pytest-cov, then do something like this until you've got every line covered:

```
# assuming it goes in napari/utils/events/evented_model
pytest napari/utils/events/_tests/test_evented_model.py  -v --cov=napari.utils.events.evented_model --cov-report=html
open htmlcov/index.html
```

**Comment by sofroniewn** [comment]
Ok, I have expanded the test - now at 94% coverage for the `EventedModel`. The only two untested lines are a fallback if field is not an event. I copied the testing approach for the evented dataclass mocking the event emission. I'm pretty happy with it. Coverage is at 100% for the `Array`.

We're getting a `minreq` fail for the `theme` changing - but I suspect it is because we're bumping the min PySide2 req to `5.13.1`, which isn't actually good. I might do some exploration with `5.13.2`. See comment here https://github.com/napari/napari/pull/2042#issuecomment-753519807 on why we need change. I'll work on fixing that but otherwise this is ready for review @jni @tlambert03.

It would also be great to your take on this @Carreau - note that we aren't actually using the `EventedModel` yet - we want to get this safely merged first, but then #2042 and #2066 would allow us to start using it (which we'd update after this PR merges)

File: napari/utils/events/_tests/test_evented_model.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+
+def test_creating_empty_evented_model():
+    """Test creating an empty evented pydantic model."""
+    model = EventedModel()
+    assert model is not None
+    assert model.events is not None
+
+
+def test_evented_model():
```
Comment (tlambert03) [inline napari/utils/events/_tests/test_evented_model.py:19]: lovely :)

**Review by tlambert03 — APPROVED** [review]
bring it!

File: setup.cfg [hunk 2]
```diff
@@ -70,7 +71,7 @@ install_requires =
 
 [options.extras_require]
 pyside2 =
-    PySide2>=5.12.3,!=5.15.0
+    PySide2>=5.13.2,!=5.15.0
```
Comment (jni) [inline setup.cfg:74]: Is this necessary? e.g. conda-forge pyqt is still 5.12.3.

Comment (jni) [reply setup.cfg:74]: Ah, just saw the discussion above. Hmm. Well, I guess it's fine since on conda-forge we depend on pyqt... 🤷

**Review by jni — APPROVED** [review]
🚀

**Pull request merged**